src/coordinator/functions/logical/or.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package logical
+
+import (
+	"fmt"
+
+	"github.com/m3db/m3db/src/coordinator/block"
+	"github.com/m3db/m3db/src/coordinator/executor/transform"
+	"github.com/m3db/m3db/src/coordinator/parser"
+)
+
+//TODO: generalize logical functions?
+const (
+	// OrType uses all values from left hand side, and appends values from the right hand side which do
+	// not have corresponding tags on the right
+	OrType = "or"
+)
+
+// NewOrOp creates a new Or operation
+func NewOrOp(lNode parser.NodeID, rNode parser.NodeID, matching *VectorMatching) BaseOp {
+	return BaseOp{
+		OperatorType: OrType,
+		LNode:        lNode,
+		RNode:        rNode,
+		Matching:     matching,
+		ProcessorFn:  NewOrNode,
+	}
+}
+
+// OrNode is a node for And operation
+type OrNode struct {
+	op         BaseOp
+	controller *transform.Controller
+}
+
+// NewOrNode creates a new OrNode
+func NewOrNode(op BaseOp, controller *transform.Controller) Processor {
+	return &OrNode{
+		op:         op,
+		controller: controller,
+	}
+}
+
+// Process processes two logical blocks, performing Or operation on them
+func (c *OrNode) Process(lhs, rhs block.Block) (block.Block, error) {
+	builder, err := c.controller.BlockBuilder(lhs.Meta(), lhs.SeriesMeta())
+	if err != nil {
+		return nil, err
+	}
+	missingIndices := c.missing(lhs.SeriesMeta(), rhs.SeriesMeta())
+	numMissing := len(missingIndices)
+	stepCount := numMissing + lhs.StepCount()
+	lIter, rIter := lhs.StepIter(), rhs.StepIter()
+	if err := builder.AddCols(stepCount); err != nil {
+		return nil, err
+	}
+
+	for index := 0; lIter.Next(); index++ {
+		lStep := lIter.Current()
+		lValues := lStep.Values()
+		for _, value := range lValues {
+			builder.AppendValue(index, value)
+		}
+	}
+
+	index := 0
+	for _, missingIndex := range missingIndices {
```

Comment:
i think this can be simplified a bit:
```
for index:=0; rIter.Next();{
rStep := rIter.Current()
rValues := rStep.Values()
for _, idx := range missingIndices {
 builder.AppendValue(index, rValues[idx])
}
}
```

Reply:
I think that this would result in different behaviour to what I'm looking for though; this would take the `missingIndex`-th values from each array in the iterator instead

Reply:
Nah my bad your suggestion is correct, got confused by column builder stuff haha